test-harness/src/test/groovy/com/netflix/counductor/integration/test/DoWhileSpec.groovy

```diff
@@ -0,0 +1,367 @@
+import com.netflix.conductor.test.util.WorkflowTestUtil
+import com.netflix.conductor.tests.utils.TestModule
+import com.netflix.governator.guice.test.ModulesForTesting
+import spock.lang.Specification
+
+import javax.inject.Inject
+
+@ModulesForTesting([TestModule.class])
+class DoWhileSpec extends Specification {
+
+    @Inject
+    ExecutionService workflowExecutionService
+
+    @Inject
+    MetadataService metadataService
+
+    @Inject
+    WorkflowExecutor workflowExecutor
+
+    @Inject
+    WorkflowTestUtil workflowTestUtil
+
+    @Inject
+    QueueDAO queueDAO
+
+    def setup() {
+        workflowTestUtil.registerWorkflows("do_while_integration_test.json",
+                "do_while_multiple_integration_test.json")
+    }
+
+    def cleanup() {
+        workflowTestUtil.clearWorkflows()
+    }
+
+    def "Test workflow with a single iteration Do While task"() {
+        given: "Number of iterations of the loop is set to 1"
+        def workflowInput = new HashMap()
+        workflowInput['loop'] = 1
+
+        when: "The workflow is started"
+        def workflowInstanceId = workflowExecutor.startWorkflow("Do_While_Workflow", 1, "looptest", workflowInput, null, null)
+
+        and: "The workflow is retrieved"
+        def workflow = workflowExecutionService.getExecutionStatus(workflowInstanceId, true)
+
+        then: "Verify that workflow is running"
+        workflow
+        workflow.status == Workflow.WorkflowStatus.RUNNING
+        workflow.tasks.size() == 2
+
+        when: "Polling for first task"
+        def task = workflowExecutionService.getLastPollTask("integration_task_0", "integration.test.worker", null)
+
+        then: "Verify that the expected task is polled"
+        task
+        task.getReferenceTaskName().endsWith(TaskUtils.getLoopOverTaskRefNameSuffix(task.getIteration()))
+
+        when: "The task is completed"
+        task.status = Task.Status.COMPLETED
+        workflowExecutionService.updateTask(task)
+        workflow = workflowExecutionService.getExecutionStatus(workflowInstanceId, true)
+
+        then: "Verify that workflow is running"
+        workflow
+        workflow.status == Workflow.WorkflowStatus.RUNNING
+        workflow.tasks.size() == 6
+
+        when: "Polling for second task"
+        task = workflowExecutionService.getLastPollTask("integration_task_1", "integration.test.worker", null)
+
+        then: "Verify that the expected task is polled"
+        task
+        task.getReferenceTaskName().endsWith(TaskUtils.getLoopOverTaskRefNameSuffix(task.getIteration()))
+
+        when: "The task is completed"
+        task.status = Task.Status.COMPLETED
+        workflowExecutionService.updateTask(task)
+        workflow = workflowExecutionService.getExecutionStatus(workflowInstanceId, true)
+
+        then: "Verify that workflow is running"
+        workflow
+        workflow.status == Workflow.WorkflowStatus.RUNNING
+        workflow.tasks.size() == 6
```

Comment:
nit: Validate the contents of the `task[]` to to verify that previous performed task completion has an affect and as a result the next task in schedule state.

Reply:
validated